hibernate-core/src/main/java/org/hibernate/type/spi/MappingContext.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.type.spi;
```

Comment:
Should it maybe just go into `org.hibernate.type`, since that package uses it?

Reply:
What I mean is: with this design, we're leaking an SPI type onto what is in principle API.

Now, yeah, I realize that the stuff in `org.hibernate.type` is been gradually reinterpreted as SPI over the years, but even if it is now SPI, it's still OK for `MappingContext` to live right there alongside its main usages.